docs/source/common/lightning_cli.rst

```diff
@@ -376,6 +376,46 @@ Note that the config object :code:`self.config` is a dictionary whose keys are g
 has the same structure as the yaml format described previously. This means for instance that the parameters used for
 instantiating the trainer class can be found in :code:`self.config['trainer']`.
 
+.. tip::
+
+    Have a look at the :class:`~pytorch_lightning.utilities.cli.LightningCLI` class API reference to learn about other
+    methods that can be extended to customize a CLI.
+
+
+Configurable callbacks
+~~~~~~~~~~~~~~~~~~~~~~
+
+As explained previously any callback can be added by including it in the config via :code:`class_path` and
+:code:`init_args` entries. However, there are other cases in which a callback should always be present and be
+configurable. This can be implemented as follows:
+
+.. testcode::
+
+    from pytorch_lightning.callbacks import EarlyStopping
+    from pytorch_lightning.utilities.cli import LightningCLI
+
+    class MyLightningCLI(LightningCLI):
+
+        def add_arguments_to_parser(self, parser):
+            parser.add_lightning_class_args(EarlyStopping, 'early_stopping')
```

Comment:
here it makes it look like this would work with anything, not just callbacks.

Reply:
Yes, it is not just for callbacks. Can be used to add any subclass of Trainer, LightningModule, LightningDataModule, Callback. There is also `parser.add_class_arguments` which can be any class, not just the lightning ones. In both cases adding arguments for more classes to the parser also needs implementation on how to use them. Only in the callback case nothing else is needed since it is just appending to the callbacks list.